File: src/main/java/io/vproxy/vfx/util/algebradata/DoubleData.java
package io.vproxy.vfx.util.algebradata;

public class DoubleData implements AlgebraData<DoubleData> {
    public final double value;

    public DoubleData(double value) {
        this.value = value;
    }

    @Override
    public DoubleData plus(DoubleData other) {
        return new DoubleData(value + other.value);
    }

    @Override
    public DoubleData minus(DoubleData other) {
        return new DoubleData(value - other.value);
    }

    @Override
    public DoubleData multiply(double v) {
        return new DoubleData(value * v);
    }

    @Override
    public DoubleData dividedBy(double v) {
        return new DoubleData(value / v);
    }

    @Override
    public String toString() {
        return "DoubleData(" + value + ")";
    }
}

File: src-ts/src/util/algebradata/DoubleData.ts
import { AlgebraData } from './AlgebraData.js';

export class DoubleData implements AlgebraData<DoubleData> {
  constructor(public readonly value: number) {}

  plus(o: DoubleData): DoubleData { return new DoubleData(this.value + o.value); }
  minus(o: DoubleData): DoubleData { return new DoubleData(this.value - o.value); }
  multiply(v: number): DoubleData { return new DoubleData(this.value * v); }
  dividedBy(v: number): DoubleData { return new DoubleData(this.value / v); }

  toString(): string { return `DoubleData(${this.value})`; }
}
